src/Orleans.Connections.Security/Security/TlsServerConnectionMiddleware.cs

```diff
@@ -151,35 +152,49 @@ private async Task InnerOnConnectionAsync(ConnectionContext context)
                         };
                     }
 
-                    var sslOptions = new SslServerAuthenticationOptions
+                    var sslOptions = new TlsServerAuthenticationOptions
                     {
                         ServerCertificate = _certificate,
                         ServerCertificateSelectionCallback = selector,
                         ClientCertificateRequired = certificateRequired,
                         EnabledSslProtocols = _options.SslProtocols,
                         CertificateRevocationCheckMode = _options.CheckCertificateRevocation ? X509RevocationMode.Online : X509RevocationMode.NoCheck,
-                        ApplicationProtocols = new List<SslApplicationProtocol>()
                     };
 
                     _options.OnAuthenticateAsServer?.Invoke(context, sslOptions);
 
-                    await sslStream.AuthenticateAsServerAsync(sslOptions, CancellationToken.None);
+                    await sslStream.AuthenticateAsServerAsync(
+                        sslOptions.ServerCertificate,
+                        sslOptions.ClientCertificateRequired,
+                        sslOptions.EnabledSslProtocols,
+                        sslOptions.CertificateRevocationCheckMode == X509RevocationMode.Online);
```

Comment:
I have some reservations around this. Specifically, there are other options on `SslServerAuthenticationOptions` which we aren't reflecting in this call, so the user misses out on the opportunity to access those options.

Reply:
The new approach allows users to modify the underlying options type if they want (on .NET Core only, since .NET Standard 2.0 does not support it)